cleverhans/attacks.py

```diff
@@ -514,6 +514,115 @@ def parse_params(self, eps=2.0, num_iterations=1, xi=1e-6, clip_min=None,
         return True
 
 
+class CarliniWagnerL2(Attack):
```

Comment:
Given that you will make PRs for the L0 and Linf attacks, is it possible to make the class general enough, that it can take a norm argument and call the right underlying algorithm?  Or is that not possible because the attacks have very different interfaces for different norms?